### PR TITLE
Add tx result

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Returning execution result in TransactionResult.
Updated nearlib to return last non-empty result.

Also it depends on https://github.com/nearprotocol/nearcore/pull/809

### CODE REVIEW

File: nearlib/near.js [hunk 1]
```diff
@@ -201,6 +201,18 @@ class Near {
...
             }
             if (result.status == 'Completed') {
+                for (j = result.logs.length - 1; j >= 0; --j) {
+                    let r = result.logs[j];
+                    if (r.result && r.result.length > 0) {
+                        result.lastResultUnparsed = r.result;
+                        try {
+                            result.lastResult = JSON.parse(Buffer.from(r.result).toString());
+                        } catch (e) {
+                            // can't parse
```
Comment (bowenwang1996) [inline nearlib/near.js]: Should we log the error?

File: node/runtime/src/lib.rs [hunk 2]
```diff
@@ -784,9 +787,7 @@ impl Runtime {
                 }
             };
         } else {
-            // wrong receipt
-            result.status = TransactionStatus::Failed;
-            result.logs.push("receipt sent to the wrong shard".to_string());
+            unreachable!("receipt sent to the wrong shard");
```
Comment (bowenwang1996) [inline node/runtime/src/lib.rs]: I don't think should be unreachable. Could be a malicious behavior.

Comment (evgenykuzyakov) [reply node/runtime/src/lib.rs]: How would you react to this malicious behavior? If it's possible to spoof receipts, then there is no security of the network.
It's either should be panic or unreachable.

File: nearlib/near.js (hunk 1 above)
Comment (evgenykuzyakov) [reply nearlib/near.js]: For non-AS JSON results, we can still parse raw data. E.g. using lastResultUnparsed

File: node/runtime/src/lib.rs (hunk 2 above)
Comment (bowenwang1996) [reply node/runtime/src/lib.rs]: I see

File: nearlib/near.js (hunk 1 above)
Comment (bowenwang1996) [reply nearlib/near.js]: Okay

**Pull request merged**